src/menu/menu_functions.rs

```diff
@@ -354,6 +357,27 @@ pub fn can_partially_complete(values: &[Suggestion], editor: &mut Editor) -> boo
     }
 }
 
+/// Split suggestion based on the display width of the matched part
+///
+/// For highlighting prefix-matched suggestions
+pub fn split_suggestion(sugg: &str, match_width: usize) -> (&str, &str) {
+    let mut match_end = sugg.len();
+    let mut curr_width = 0;
+    for (offset, grapheme) in sugg.grapheme_indices(true) {
+        if curr_width >= match_width {
+            match_end = offset;
+            break;
+        }
+        // Strip quotes from the beginning
+        if offset == 0 && (grapheme == "`" || grapheme == "'" || grapheme == "\"") {
+            continue;
+        }
+        curr_width += grapheme.width();
```

Comment:
I guess with this code doing operations on Unicode width would not panic.

But I have a question if this misbehaves if the user tries to complete starting from `"` etc. or `"` being a particular suggestion? (Is this relying on a particular behavior of our `Completer` implementations to take out all quotes?)

Reply:
I was planning on allowing adding a separate display value on `Suggestion`s, so that when showing suggestions for files, the displayed values could be unquoted and unescaped. Then, Reedline would no longer have to strip quotes. This current quote stripping is really for Nushell, and I assume it won't hurt other users of Reedline either.